### ISSUE TITLE
Returning tree_obj_values in place of tree_obj in spider.py

### ISSUE THREAD

**Issue by awasthiabhijeet** [OhadRubin/SmBop#43]
Hi @OhadRubin ,

If value_pred is True, would it be correct to simply return `tree_obj_values` in place of `tree_obj` in the MetaField?
https://github.com/OhadRubin/SmBop/blob/e7a6fce7af5aa5545bd3cfca6c4c4dbef610cd6b/smbop/dataset_readers/spider.py#L271

Thanks,

**Comment by OhadRubin** [comment]
I'm not sure, can you try and let me know?

**Comment by awasthiabhijeet** [comment]
Sure, I will comment here in case I try it.
I actually needed it for processing non-spider SQLs, but I could figure out an alternate way that does not need this.

It would be great if you could enable a discussion thread for this repo :)
Could you also provide some insights around why softmax loss is being used at each level, even though this looks like a multi-label classification?
There is a flag for using bce_loss, but it's not present in the implementation. 
Is it better to use softmax than using bce_loss?

**Comment by OhadRubin** [comment]
Yeah, we tried bce loss and it was worse.
My intuition on why softmax works better is that it acts as a regularizer, otherwise for bce, there is nothing constraining the logits to stay in the same numerical range.

**Comment by awasthiabhijeet** [comment]
> Hi @OhadRubin ,
> 
> If value_pred is True, would it be correct to simply return `tree_obj_values` in place of `tree_obj` in the MetaField?
> https://github.com/OhadRubin/SmBop/blob/e7a6fce7af5aa5545bd3cfca6c4c4dbef610cd6b/smbop/dataset_readers/spider.py#L271
> 
> Thanks,

Update: Using `tree_obj_values` in place of `tree_obj` seems to work fine when `value_pred` is set to `True`
